source/particle/interpolator/quadratic_least_squares.cc

```diff
@@ -0,0 +1,294 @@
+            A[2][positions_index] = relative_particle_position[1];
+            if (dim == 2)
+              {
+                A[3][positions_index] = relative_particle_position[0] * relative_particle_position[1];
+                A[4][positions_index] = std::pow(relative_particle_position[0], 2);
+                A[5][positions_index] = std::pow(relative_particle_position[1], 2);
+              }
+            else
+              {
+                A[3][positions_index] = relative_particle_position[2];
+                A[4][positions_index] = relative_particle_position[0] * relative_particle_position[1];
+                A[5][positions_index] = relative_particle_position[0] * relative_particle_position[2];
+                A[6][positions_index] = relative_particle_position[1] * relative_particle_position[2];
+                A[7][positions_index] = std::pow(relative_particle_position[0], 2);
+                A[8][positions_index] = std::pow(relative_particle_position[1], 2);
+                A[9][positions_index] = std::pow(relative_particle_position[2], 2);
+              }
+          }
+
+        for (unsigned int column_index = 0; column_index < n_matrix_columns; ++column_index)
+          qr.append_column(A[column_index]);
+        AssertThrow(qr.size() == n_matrix_columns,
+                    ExcMessage("The matrix A was rank deficent during quadratic least squares interpolation."));
+        // If A is rank deficent, qr.append_column will not append the column.
+        // We check that all columns were added through this assertion
+
+        for (unsigned int property_index = 0; property_index < n_particle_properties; ++property_index)
+          {
+            if (selected_properties[property_index])
+              {
+                qr.multiply_with_QT(QTb[property_index], b[property_index]);
+                qr.solve(c[property_index], QTb[property_index]);
+              }
+          }
+        unsigned int index_positions = 0;
+        for (typename std::vector<Point<dim>>::const_iterator itr = positions.begin(); itr != positions.end(); ++itr, ++index_positions)
+          {
+            const Tensor<1, dim, double> relative_support_point_location = (*itr - approximated_cell_midpoint) / cell_diameter;
+            for (unsigned int property_index = 0; property_index < n_particle_properties; ++property_index)
+              {
+                double interpolated_value = c[property_index][0] +
+                                            c[property_index][1] * relative_support_point_location[0] +
+                                            c[property_index][2] * relative_support_point_location[1];
+                if (dim == 2)
+                  {
+                    interpolated_value += c[property_index][3] * relative_support_point_location[0] * relative_support_point_location[1] +
+                                          c[property_index][4] * std::pow(relative_support_point_location[0], 2) +
```

Comment:
`std::pow` is convenient, but it is often much slower than `relative_support_point_location[0]*relative_support_point_location[0]` (because it can calculate arbitrary powers). Maybe there are specializations for the power of 2, but I would not count on that. Please replace occurences of std::pow by actual multiplications.
```suggestion
                                          c[property_index][4] * std::pow(relative_support_point_location[0], 2) +
```